src/main/java/digital/slovensko/autogram/core/eforms/XDCBuilder.java

```diff
@@ -69,51 +62,19 @@ private XDCTransformer(String identifier, String xsdSchema, String xsltSchema, S
         this.mediaDestinationTypeDescription = mediaDestinationTypeDescription;
     }
 
-    public static XDCTransformer buildFromSigningParametersAndDocument(SigningParameters sp, DSSDocument document) throws InvalidXMLException {
-        try {
-            var builderFactory = DocumentBuilderFactory.newInstance();
-            builderFactory.setNamespaceAware(true);
-            builderFactory.setFeature("http://xml.org/sax/features/external-general-entities", false);
-
-            return new XDCTransformer(sp.getSchema(),
-                    sp.getTransformation(),
-                    sp.getPropertiesCanonicalization(),
-                    sp.getDigestAlgorithm(),
-                    builderFactory.newDocumentBuilder().parse(new InputSource(document.openStream())));
-        } catch (Exception e) {
-            throw new InvalidXMLException("XML Datacontainer validation failed", "Unable to process document");
-        }
-    }
-
-    private XDCTransformer(String xsdSchema, String xsltSchema, String canonicalizationMethod,
-                          DigestAlgorithm digestAlgorithm, Document document) {
-        this.xsdSchema = xsdSchema;
-        this.xsltSchema = xsltSchema;
-        this.canonicalizationMethod = canonicalizationMethod;
-        this.digestAlgorithm = digestAlgorithm;
-        this.identifierUri = null;
-        this.identifierVersion = null;
-        this.containerXmlns = null;
-        this.mediaDestinationTypeDescription = null;
-        this.document = document;
-    }
-
     public DSSDocument transform(DSSDocument dssDocument) {
```

Comment:
Tu je mozno otazka, ze preco tato trieda xdcbuilder vobec existuje. Ved by nam stacila jedna metoda/funkcia co dostane signing parametre a document a urobi co treba nie?

Reply:
spravil som z toho abstract classu. Neviem, či by som to úplne hádzal do jeden funkcie celé. Nechal by som takto v osobitnom súbore, kým to funguje. Aj keď je pravda, že teraz je tam pomerne veľa zbytočne malých metód.